lib/cart/cart_repository.dart

```diff
@@ -35,101 +35,66 @@ class CartRepository {
 
   /// The latest Cart in the stream
   Cart get latestCart => _cartStreamController.value;
+  bool get hasCart => _cartStreamController.hasValue;
 
-  Future<bool> addProductToCart({
-    required String productCode,
+  Future<void> addProductToCart({
+    required Product product,
     required int quantity,
   }) async {
     const cartCode = kTestCart;
 
-    try {
-      await _cartApi.addProductToCart(
-        cartCode: cartCode,
-        productCode: productCode,
-        quantity: quantity,
-      );
-
-      _cartMessageStreamController.add(
-        AddToCartMessage(
-          CartMessageType.success,
-          'Added $quantity of $productCode to cart',
-        ),
-      );
-
-      final cart = await _cartApi.fetchCart(cartCode: kTestCart);
-      _cartStreamController.add(cart);
-
-      return true;
-    } on DioException catch (e) {
-      _cartMessageStreamController.add(
-        AddToCartMessage(
-          CartMessageType.error,
-          'Failed to add $quantity of $productCode to cart, with $e',
-        ),
-      );
```

Comment:
I think we are mixing different responsibilities here, the event should only reflect what happened. 

The user has added 3 items of product xxx to the cart 👉 we fire a event "Add to cart" with product xxx and quantity 3. And that's it. The rest (error feedback, logging, and in my case: analytics) should be handled elsewhere.

So I went a bit rogue and simplify the methods in this repository (return type, error management, ...)